okhttp/src/test/java/okhttp3/internal/tls/ClientAuthTest.java

```diff
@@ -234,7 +238,7 @@ public void setUp() {
   @Test public void invalidClientAuthFails() throws Throwable {
     // TODO https://github.com/square/okhttp/issues/4598
     // StreamReset stream was reset: PROT...
-    assumeFalse(getJvmSpecVersion().matches("1[123]"));
+    //jdkMatchRule.expectFailure(fromMajor(11), anything());
```

Comment:
I think I want different syntax in tests, even if it makes the JdkMatchRule more complex. Maybe this?

```
  @Rule public final JdkMatchRule jdkMatchRule = new JdkMatchRule();

  @Test public void invalidClientAuthFails() throws Throwable {
    jdkMatchRule.assumeJdkLessThan(12)
  }
```

Reply:
The syntax of Hamcrest matchers seems so awkward

Reply:
I specifically designed to not use assumptions.  I wanted failures and to verify the failures on a secondary path, such that tests are not skipped or flaky, but understood including when they start passing because of fixes upstream.